### DISCUSSION TITLE
ToolTips should be max 200 characters

### DISCUSSION THREAD

**Discussion by jwikman** [StefanMaron/BusinessCentral.LinterCop#347] [New Rule]
According to the guidelines for ToolTips (https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/user-assistance#guidelines-for-tooltip-text), a ToolTip should not exceed 200 characters including spaces.

The rule should check that the ToolTip, if specified, has less or equal to 200 characters including spaces.

**Comment by Arthurvdv** [comment]
Added this to the existing LC0026 rule and is now available in the (pre)release version [v0.30.8](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.8).

**Reply by Arthurvdv** [reply]
When I executed the prelease on some of our projects this one get's quite a lot of hits 😯

In the documentation the Account Type field is an great example of exceeding this limit is justified. It feels not right to have to set a pragma around cases like this. Do we set the bar higher, like 500, 1000 , ..? Or do not include this check in the LC0026?

**Reply by jwikman** [reply]
I found a few as well where longer tooltips were justified.

Maybe this one should not be included in the LC0026, since this one might be justified in a way that probably not the other checks are.

**Reply by jwikman** [reply]
If this was a rule for itself, I think I would have it as an "Info" in our projects, and the rest of the LC0026 as warnings (and we do not allow warnings to ship with the product).

**Reply by jwikman** [reply]
Ideally the recommended maximum number of characters could be configurable in the lintercop.json settings file, but it might be overkill. 🙂

**Comment by jwikman** [comment]
I talked to our "ToolTip Expert" about this, and we agreed on that this should be a rule of its own instead of a part of LC0026.
The other ToolTip rules are hard rules, but this one is more of a recommendation one. 

Does it make sense?

**Reply by Arthurvdv** [reply]
I'm thinking of going to other way around: Restore the LC0026 to only focus on ending dot and create a new rule that combines all the guidelines as provided in the docs. Does that work for you?

**Reply by jwikman** [reply]
I think that the "ending with a dot", "fields should start with 'specifies'" and "Do not use line breaks" are all, more or less, mandatory ToolTip rules, whereas ToolTip length is a softer recommendation (Try to...).

I would vote for one of these two ways of handling these:
1. Separate rules for all four. In that way you can configure the separate recommendation per your own taste.
2. The first three in LC0026 and a new rule for length recommendation. In this way you can configure to skip the length rule if you do not care.

The first one is the most flexible one, but I guess that it comes with a price in terms of performance?

**Reply by Arthurvdv** [reply]
I'm going to follow your lead on the first option.

You're right about that's the most flexible one, so everyone can tailor these rules which they feel is a best fit. This doesn't influence performance, as they where already four rules (where they previously had the same identifier). Thank you for taking the time for these suggestions here. 

The updated prerelease version [v0.30.8](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.8) has now three new rules :-)

**Reply by jwikman** [reply]
Thanks @Arthurvdv, it looks very good!

Should we close this discussion as resolved then? 🙂

**Reply by Arthurvdv** [reply]
Good idea, I'll close the both of them.
